### PR TITLE
[ISSUE #3228] Fix unexpected result of loadbalance select

### PR DESCRIPTION
**Pull request by ywengineer**
// Describe your PR here; eg. Fixes #issueNo

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by qicz** [comment]
@ywengineer I toggle ci again. wait a moment.

File: shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java [hunk 1]
```diff
@@ -40,10 +41,15 @@ public Upstream select(final List<Upstream> upstreamList, final String ip) {
         if (CollectionUtils.isEmpty(upstreamList)) {
             return null;
         }
-        if (upstreamList.size() == 1) {
-            return upstreamList.get(0);
+        // available upstream list
+        List<Upstream> available = upstreamList.stream().filter(upstream -> getWeight(upstream) > 0).collect(Collectors.toList());
```
Comment (KevinClair) [inline shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java:45]: If there are three upstreams in `DividePlugin`, and their weight is 0, maybe it's error here, right?

Comment (ywengineer) [reply shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java:45]: Yes

Comment (yu199195) [reply shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java:45]: @ywengineer  this pr need update？

Comment (ywengineer) [reply shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java:45]: no, return null if there isn't any available upstream, see line 47.

Comment (yu199195) [reply shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java:45]: if used HashLoadBalancer？ it not need weight~

**Review by yu199195 — CHANGES_REQUESTED** [review]
not merged

File: shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java (hunk 1 above)
Comment (ywengineer) [reply shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java:45]: so, do you mean that we should use the status of upstream to determine it is available or not is better than weight?

**Comment by ywengineer** [comment]
Regardless of any LoadBalancer, closed upstream should not be used. is it right? 
Do you mean that we should use the status of upstream to determine it is available or not is better than weight? am i right?

**Pull request closed without merging**